File: src/app/src/main/java/com/huawei/cardreader/java/scannedcardlist/model/QRcodmodel.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.huawei.cardreader.java.scannedcardlist.model;

import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

import com.huawei.cardreader.java.scannedcardlist.roomdb.BusinessCardEntity;

import java.util.List;

/**
 * The type Q rcodmodel.
 */
public class QRcodmodel implements Parcelable {
    private Bitmap bitmap;
    private List<BusinessCardEntity> businesscardUserDetailsList;

    /**
     * Instantiates a new Q rcodmodel.
     */
    public QRcodmodel() {
    }


    /**
     * Gets bitmap.
     *
     * @return the bitmap
     */
    public Bitmap getBitmap() {
        return bitmap;
    }

    /**
     * Sets bitmap.
     *
     * @param bitmap the bitmap
     */
    public void setBitmap(Bitmap bitmap) {
        this.bitmap = bitmap;
    }

    /**
     * Gets businesscard user details list.
     *
     * @return the businesscard user details list
     */
    public List<BusinessCardEntity> getBusinesscardUserDetailsList() {
        return businesscardUserDetailsList;
    }

    /**
     * Sets businesscard user details list.
     *
     * @param businesscardUserDetailsList the businesscard user details list
     */
    public void setBusinesscardUserDetailsList(List<BusinessCardEntity> businesscardUserDetailsList) {
        this.businesscardUserDetailsList = businesscardUserDetailsList;
    }

    /**
     * Instantiates a new Q rcodmodel.
     *
     * @param in the in
     */
    protected QRcodmodel(Parcel in) {
        bitmap = in.readParcelable(Bitmap.class.getClassLoader());
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(bitmap, flags);
    }

    @Override
    public int describeContents() {
        return 0;
    }


    /**
     * The constant CREATOR.
     */
    public static final Creator<QRcodmodel> CREATOR = new Creator<QRcodmodel>() {
        @Override
        public QRcodmodel createFromParcel(Parcel in) {
            return new QRcodmodel(in);
        }

        @Override
        public QRcodmodel[] newArray(int size) {
            return new QRcodmodel[size];
        }
    };

}

File: src/app/src/main/java/com/huawei/cardreader/kotlin/scannedcardlist/model/QRcodmodel.kt
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.huawei.cardreader.kotlin.scannedcardlist.model

import android.graphics.Bitmap
import android.os.Parcel
import android.os.Parcelable
import com.huawei.cardreader.kotlin.scannedcardlist.roomdb.BusinessCardEntity

/**
 * The type Q rcodmodel.
 */
class QRcodmodel : Parcelable {
    /**
     * Gets bitmap.
     *
     * @return the bitmap
     */
    /**
     * Sets bitmap.
     *
     * @param bitmap the bitmap
     */
    var bitmap: Bitmap? = null
    /**
     * Gets businesscard user details list.
     *
     * @return the businesscard user details list
     */
    /**
     * Sets businesscard user details list.
     *
     * @param businesscardUserDetailsList the businesscard user details list
     */
    var businesscardUserDetailsList: List<BusinessCardEntity>? = null

    /**
     * Instantiates a new Q rcodmodel.
     */
    constructor() {}

    /**
     * Instantiates a new Q rcodmodel.
     *
     * @param in the in
     */
    protected constructor(`in`: Parcel) {
        bitmap = `in`.readParcelable(Bitmap::class.java.classLoader)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(bitmap, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        /**
         * The constant CREATOR.
         */
        val CREATOR: Parcelable.Creator<QRcodmodel?> = object : Parcelable.Creator<QRcodmodel?> {
            override fun createFromParcel(`in`: Parcel): QRcodmodel? {
                return QRcodmodel(`in`)
            }

            override fun newArray(size: Int): Array<QRcodmodel?> {
                return arrayOfNulls(size)
            }
        }
    }
}